### PR TITLE
BUG: Fix mutidimensional model cov_params when using pandas

### LINKED ISSUE #814: MNLogit.cov_params is broken when wrapped.
Straight forward. We will need a wrap2d function for this (also VAR and systems of equations).

### LINKED ISSUE #2270: BUG: VAR cov_params pandas wrapper broken
from example in vector_ar tutorial

```
In [34]: results._results.cov_params.shape
Out[34]: (36, 36)

In [37]: results.cov_params

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-37-14357748fa96> in <module>()

<snip>
ValueError: Shape of passed values is (36, 36), indices imply (12, 12)
```

I assume this is similar to the problem in MNLogit that #2197 should fix

### LINKED ISSUE #5305: BUG: VAR cov_params pandas wrapper is broken
Hi,

I get error message for some models when I try to compute the standard errors for the coefficients. But the code in the package seems to be working well. Can you guys help me? Thanks!

`z=result_bp.ys_lagged`
`cov=np.kron(scipy.linalg.inv(np.dot(z.T, z)), result_bp.sigma_u)`
`stderr = np.sqrt(np.diag(cov))`
`stderr.reshape((48, 4), order='C')`

This one works fine if I manually input (48,4). The following is the error message when I try to use the attribute (stderr) in VAR.

>ValueError                                Traceback (most recent call last)
~\Anaconda3\lib\site-packages\pandas\core\internals.py in create_block_manager_from_blocks(blocks, axes)
   4858 
-> 4859         mgr = BlockManager(blocks, axes)
   4860         mgr._consolidate_inplace()

>~\Anaconda3\lib\site-packages\pandas\core\internals.py in __init__(self, blocks, axes, do_integrity_check)
   3281         if do_integrity_check:
-> 3282             self._verify_integrity()
   3283 

>~\Anaconda3\lib\site-packages\pandas\core\internals.py in _verify_integrity(self)
   3492             if block._verify_integrity and block.shape[1:] != mgr_shape[1:]:
-> 3493                 construction_error(tot_items, block.shape[1:], self.axes)
   3494         if len(self.items) != tot_items:

>~\Anaconda3\lib\site-packages\pandas\core\internals.py in construction_error(tot_items, block_shape, axes, e)
   4842     raise ValueError("Shape of passed values is {0}, indices imply {1}".format(
-> 4843         passed, implied))
   4844 

>ValueError: Shape of passed values is (4, 48), indices imply (4, 24)

>During handling of the above exception, another exception occurred:

>ValueError                                Traceback (most recent call last)
<ipython-input-110-593b97e3e723> in <module>()
----> 1 result_bp.stderr

>~\Anaconda3\lib\site-packages\statsmodels\base\wrapper.py in __getattribute__(self, attr)
     39             obj = data.wrap_output(obj, how[0], *how[1:])
     40         elif how:
---> 41             obj = data.wrap_output(obj, how=how)
     42 
     43         return obj

>~\Anaconda3\lib\site-packages\statsmodels\base\data.py in wrap_output(self, obj, how, names)
    416             return self.attach_dates(obj)
    417         elif how == 'columns_eq':
--> 418             return self.attach_columns_eq(obj)
    419         elif how == 'cov_eq':
    420             return self.attach_cov_eq(obj)

>~\Anaconda3\lib\site-packages\statsmodels\base\data.py in attach_columns_eq(self, result)
    528 
    529     def attach_columns_eq(self, result):
--> 530         return DataFrame(result, index=self.xnames, columns=self.ynames)
    531 
    532     def attach_cov(self, result):

>~\Anaconda3\lib\site-packages\pandas\core\frame.py in __init__(self, data, index, columns, dtype, copy)
    377             else:
    378                 mgr = self._init_ndarray(data, index, columns, dtype=dtype,
--> 379                                          copy=copy)
    380         elif isinstance(data, (list, types.GeneratorType)):
    381             if isinstance(data, types.GeneratorType):

>~\Anaconda3\lib\site-packages\pandas\core\frame.py in _init_ndarray(self, values, index, columns, dtype, copy)
    534             values = maybe_infer_to_datetimelike(values)
    535 
--> 536         return create_block_manager_from_blocks([values], [columns, index])
    537 
    538     @property

>~\Anaconda3\lib\site-packages\pandas\core\internals.py in create_block_manager_from_blocks(blocks, axes)
   4864         blocks = [getattr(b, 'values', b) for b in blocks]
   4865         tot_items = sum(b.shape[0] for b in blocks)
-> 4866         construction_error(tot_items, blocks[0].shape[1:], axes, e)
   4867 
   4868 

>~\Anaconda3\lib\site-packages\pandas\core\internals.py in construction_error(tot_items, block_shape, axes, e)
   4841         raise ValueError("Empty data passed with indices specified.")
   4842     raise ValueError("Shape of passed values is {0}, indices imply {1}".format(
-> 4843         passed, implied))
   4844 
   4845 

>ValueError: Shape of passed values is (4, 48), indices imply (4, 24)

### PR DESCRIPTION
**Pull request by bashtage**
- [X] closes #814, closes #5305, closes #2270
- [X] tests added / passed. 
- [X] code/documentation is well formatted.  
- [X] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in master and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/master -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

**Comment by bashtage** [comment]
@josef-pkt Would be good if you could look at this since it is deep (affects base).

File: statsmodels/discrete/discrete_model.py [hunk 1]
```diff
@@ -4214,75 +4229,114 @@ def __init__(self, model, mlefit):
...
+
+
 wrap.populate_wrapper(GeneralizedPoissonResultsWrapper,
                       GeneralizedPoissonResults)
 
+
 class PoissonResultsWrapper(lm.RegressionResultsWrapper):
     pass
-    #_methods = {
+    # _methods = {
```
Comment (jbrockmendel) [inline statsmodels/discrete/discrete_model.py]: should be removed

File: statsmodels/tsa/vector_ar/tests/test_var.py [hunk 2]
```diff
@@ -803,3 +804,17 @@ def test_exog(self):
...
         getattr(bivariate_var_result, attr)
+
+
+def test_var_cov_params(bivariate_var_data):
+    df = pd.DataFrame(bivariate_var_data, columns=['x', 'y'])
+    mod = VAR(df)
+    res = mod.fit(2)
+    cov = res.cov_params()
+    assert isinstance(cov, pd.DataFrame)
+    print(cov)
```
Comment (jbrockmendel) [inline statsmodels/tsa/vector_ar/tests/test_var.py]: remove

**Comment by jbrockmendel** [comment]
Instead of "xname.yname", I think we'd be better off with a MultiIndex that didn't mangle the names at all (e.g. retained non-str names), see https://github.com/jbrockmendel/sm2/blob/master/sm2/base/data.py#L527

**Comment by bashtage** [comment]
> Instead of "xname.yname", I think we'd be better off with a MultiIndex that didn't mangle the names at all (e.g. retained non-str names)

Great idea.  Much better than munged strings.

**Comment by bashtage** [comment]
As for sm2, that fix doesn't work across the board since VAR and MNLogit use reverse orders for the outer and inner indices.

File: statsmodels/tsa/vector_ar/tests/test_var.py [hunk 3]
```diff
@@ -803,3 +805,15 @@ def test_exog(self):
 def test_deprecated_attributes_varresults(bivariate_var_result, attr):
     with pytest.warns(FutureWarning):
         getattr(bivariate_var_result, attr)
+
+
+def test_var_cov_params(bivariate_var_data):
```
Comment (jbrockmendel) [inline statsmodels/tsa/vector_ar/tests/test_var.py]: test name should reflect its about pandas.  ill handle in follow-up

File: statsmodels/tsa/vector_ar/tests/test_var.py [hunk 4]
```diff
@@ -11,6 +11,8 @@
 import sys
 
 import numpy as np
+import pandas as pd
+from pandas.testing import assert_index_equal
```
Comment (jbrockmendel) [inline statsmodels/tsa/vector_ar/tests/test_var.py]: pretty much everywhere else we import from pandas.util.testing (and im not sure if we ever use statsmodels.compat.pandas.ptesting).  ill take a look in follow-up.

File: statsmodels/discrete/discrete_model.py [hunk 5]
```diff
@@ -4213,75 +4227,108 @@ def __init__(self, model, mlefit):
...
 class L1PoissonResultsWrapper(lm.RegressionResultsWrapper):
     pass
-    #_methods = {
+    # _methods = {
     #        "predict_prob" : "rows",
     #        }
-    #_wrap_methods = lm.wrap.union_dicts(
+    # _wrap_methods = lm.wrap.union_dicts(
     #                            lm.RegressionResultsWrapper._wrap_methods,
     #                            _methods)
```
Comment (jbrockmendel) [inline statsmodels/discrete/discrete_model.py]: ill fix in follow-up

**Comment by jbrockmendel** [comment]
minor comments, generally looks good

File: statsmodels/discrete/discrete_model.py [hunk 6]
```diff
@@ -2162,7 +2162,20 @@ class MNLogit(MultinomialModel):
...
+        super(MNLogit, self).__init__(endog, exog, **kwargs)
+
+        # Override cov_names since multivariate model
+        yname = self.endog_names
+        ynames = self._ynames_map
+        ynames = MultinomialResults._maybe_convert_ynames_int(ynames)
+        # use range below to ensure sortedness
+        ynames = [ynames[key] for key in range(int(self.J))]
+        idx = MultiIndex.from_product((ynames[1:], self.data.xnames),
+                                      names=(yname, None))
```
Comment (jbrockmendel) [inline statsmodels/discrete/discrete_model.py]: thoughts on adding level names?

File: statsmodels/tsa/vector_ar/var_model.py [hunk 7]
```diff
@@ -642,7 +641,8 @@ def fit(self, maxlags=None, method='ols', ic=None, trend='c',
             self.data.xnames = (self.data.xnames[:k_trend] +
                                 x_names_to_add +
                                 self.data.xnames[k_trend:])
-
+        self.data.cov_names = pd.MultiIndex.from_product((self.data.xnames,
+                                                          self.data.ynames))
```
Comment (jbrockmendel) [inline statsmodels/tsa/vector_ar/var_model.py]: here too level names might be helpful

**Pull request merged**